### PR TITLE
Configurable SMTP Timeout (GSI-986)

### PR DESCRIPTION
**Pull request by TheByronHimes**
Adds an optional `smtp_timeout` config option. By default it will ~maintain the prior infini-hang behaviour~ time out after 60 seconds. Added an error class to the `SMTPClient` to catch any instance of `OSError` raised by a rejected connection or timeout. 

Added in some debug statements in the SMTP module to aid in future bug hunting. 

Bumped the version from `2.0.0` to `2.1.0`.

### CODE REVIEW

File: src/ns/adapters/outbound/smtp_client.py [hunk 1]
```diff
@@ -51,6 +51,13 @@ class SmtpClientConfig(BaseSettings):
     use_starttls: bool = Field(
         default=True, description="Boolean flag indicating the use of STARTTLS"
     )
+    smtp_timeout: PositiveInt | None = Field(
```
Comment (mephenor) [inline src/ns/adapters/outbound/smtp_client.py:54]: To be type consistent with what the `SMTP` class expects this should be `PositiveFloat` instead

**Review by mephenor — APPROVED** [review]
Looks good, just one suggestion

**Review by dontseyit — COMMENTED** [review]
Why the default is None? I would expect, as a default behavior, an error message eventually if there is a problem in connection rather than infinite waiting. Then adjusting the value accordingly if necessary. So I would say 60 seconds.

File: src/ns/adapters/outbound/smtp_client.py (hunk 1 above)
Comment (TheByronHimes) [reply src/ns/adapters/outbound/smtp_client.py]: Yes, that is the way it should be. Look here: [7991526](https://github.com/ghga-de/notification-service/pull/20/commits/7991526a5433694407a0891eff2eda7ee3ca6b10)

**Review by dontseyit — APPROVED** [review]
Looks good to me now, thanks.

**Review by mephenor — APPROVED** [review]

**Pull request merged**